integrations/http/save.go

```diff
@@ -0,0 +1,87 @@
+package http
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"go.uber.org/zap"
+)
+
+const (
+	HeaderContentType = "Content-Type"
+	ContentTypeForm   = "application/x-www-form-urlencoded"
+)
+
+// handleAuth saves a new autokitteh connection with user-submitted data.
+func (h handler) handleAuth(w http.ResponseWriter, r *http.Request) {
+	l := h.logger.With(zap.String("urlPath", r.URL.Path))
+
+	// Check "Content-Type" header.
+	ct := r.Header.Get(HeaderContentType)
+	if ct != ContentTypeForm {
+		l.Warn("Unexpected header value",
+			zap.String("header", HeaderContentType),
+			zap.String("got", ct),
+			zap.String("want", ContentTypeForm),
+		)
+		e := fmt.Sprintf("Unexpected Content-Type header: %q", ct)
+		u := fmt.Sprintf("%serror.html?error=%s", uiPath, url.QueryEscape(e))
+		http.Redirect(w, r, u, http.StatusFound)
+		return
+	}
+
+	// Read and parse POST request body.
+	err := r.ParseForm()
+	if err != nil {
+		l.Warn("Failed to parse inbound HTTP request",
+			zap.Error(err),
+		)
+		e := "Form parsing error: " + err.Error()
+		u := fmt.Sprintf("%serror.html?error=%s", uiPath, url.QueryEscape(e))
+		http.Redirect(w, r, u, http.StatusFound)
+		return
+	}
+	basic := r.Form.Get("basic_username") + ":" + r.Form.Get("basic_password")
+	bearer := r.Form.Get("bearer_access_token")
+
+	authz := ""
+	switch {
```

Comment:
do you want to error if both provided ?

Reply:
When a user uses the HTML form they can't submit more than one.

If someone want to send a handcrafted POST with both options that's their problem, I only care that they can't create damage/risks.

Reply:
It's not just `their problem`, it becomes our problem. Also there might be a security issues. But CSRF is for another PR :)